science/build_info.py

```diff
@@ -37,7 +37,7 @@ def _maybe_gather_git_state() -> str | None:
     if git_info.returncode == 0:
         return git_info.stdout.strip()
 
-    logger.warning(f"Failed to gather git state for provenance.")
+    logger.warning("Failed to gather git state for provenance.")
```

Comment:
The changes in this file and the next 5 are all due to the switch from `black` / `isort` /`autoflake` to `ruff`.